### PR TITLE
Fix edit button in a contact's profile not shown when selected from a group member list(fix #3267)

### LINKED ISSUE #3267: View Profile: "Edit Name" is not available when accessing the contact from the members section in a group view
`Edit Name` is not available when accessing the contact from the members section in a group view.
It is confusing to have 2 slightly different versions of the same view, depending on how it was opened.

### PR DESCRIPTION
**Pull request by farooqkz**
This uses `ViewProfile` component to show someone's profile when they are selected from group member list. This fixes #3267 

Note: currently, when the X button is pressed in the user's profile which is shown this way, the entire dialogue is closed. Should it show the previous dialogue instead which is the group member list?

### CODE REVIEW

File: src/renderer/components/dialogs/ViewGroup.tsx [hunk 1]
```diff
@@ -217,9 +217,15 @@ function ViewGroupInner(props: {
     null
   )
 
+  if (viewMode === 'profile' && !profileContact) {
```
Comment (Simon-Laux) [inline src/renderer/components/dialogs/ViewGroup.tsx:220]: you could just check for profile contact and remove the view mode, then you don't need this if statement and have simpler code?

**Review by Simon-Laux — COMMENTED** [review]
So the back button does not exist anymore in this pr?
Point of my issue was to have the edit button in both cases, the back button should not be changed.

File: src/renderer/components/dialogs/ViewGroup.tsx (hunk 1 above)
Comment (farooqkz) [reply src/renderer/components/dialogs/ViewGroup.tsx:220]: the thing is that if view mode is profile, profile contact must be non-null. If it's null, it's a bug. The code shows the main view anyway, but this error log is printed to console.

**Comment by farooqkz** [comment]
> So the back button does not exist anymore in this pr? Point of my issue was to have the edit button in both cases, the back button should not be changed.

How about replacing the close button with a back button?

File: src/renderer/components/dialogs/ViewGroup.tsx (hunk 1 above)
Comment (Simon-Laux) [reply src/renderer/components/dialogs/ViewGroup.tsx:219]: my idea here was to remove the view mode and make it dependent on wether `profileContact` is set or not, so one variable less

**Comment by Simon-Laux** [comment]
> How about replacing the close button with a back button?

No, this issue is only about making the edit button available in both cases, the UI should not be changed.
Also the difficulty of passing down a back button callback is the same or less effort than passing down a property to change the appearance and logic of the close button.

**Review by Simon-Laux — CHANGES_REQUESTED** [review]
as I said UI besides the edit button should not be changed, the back button should still be shown and the close button should still close the whole dialog.

**Comment by farooqkz** [comment]
> as I said UI besides the edit button should not be changed, the back button should still be shown and the close button should still close the whole dialog. I can take over this pr if you want.

Nope I will do it. I think I misread your message about both having back button and close button.

**Review by Simon-Laux — APPROVED** [review]
Great, one nitpick, if you change the name and go back the old name is still shown, but we can fix this separately and just make a bug for it for now

**Pull request merged**